jira/server/scio_search/src/main/java/ScioSearchConfigRestPlugin/impl/ScioSearchIssueSecurityMembers.java

```diff
@@ -0,0 +1,134 @@
+
+  @Inject
+  public ScioSearchIssueSecurityMembers(
+      UserManager userManager, PluginSettingsFactory pluginSettingsFactory) {
+    this.userManager = userManager;
+    this.pluginSettingsFactory = pluginSettingsFactory;
+  }
+
+  private void validateUserIsAdmin() {
+    final UserProfile profile = userManager.getRemoteUser();
+    if (profile == null || !userManager.isSystemAdmin(profile.getUserKey())) {
+      throw new UnauthorizedException("Unauthorized");
+    }
+  }
+
+  @GET
+  @Produces(MediaType.APPLICATION_JSON)
+  public ScioIssueSecurityMembersResponse setTarget(@QueryParam("schemeId") String schemeIdStr) {
+    logger.debug(String.format("Received request for getting issuesecuritymembers: %s", schemeIdStr));
+    validateUserIsAdmin();
+    Long schemeId = Long.parseLong(schemeIdStr);
+
+    final IssueSecurityLevelManager issueSecurityLevelManager = ComponentAccessor.getIssueSecurityLevelManager();
+    final IssueSecuritySchemeManager issueSecuritySchemeManager = ComponentAccessor.getComponentOfType(IssueSecuritySchemeManager.class);
+    final Collection<IssueSecurityLevel> issueSecurityLevels = issueSecurityLevelManager.getIssueSecurityLevels(schemeId);
+
+    final ScioIssueSecurityMembersResponse response = new ScioIssueSecurityMembersResponse();
+    response.startAt = 0;
+    response.isLast = true; // no response pagination
+    if (issueSecurityLevels != null) {
+      for (IssueSecurityLevel isl : issueSecurityLevels) {
+        logger.debug(isl.getDescription() + ":" + isl.getName() + ":" + String.valueOf(isl.getId()) + ":" + String.valueOf(isl.getSchemeId()));
+        List<IssueSecurityLevelPermission> issueSecurityLevelPermissions = issueSecuritySchemeManager.getPermissionsBySecurityLevel(isl.getId());
+        if (issueSecurityLevels != null) {
+          response.maxResults += 1;
+          response.total += 1;
+          response.values = new ArrayList();
+          for (IssueSecurityLevelPermission islp : issueSecurityLevelPermissions) {
+            logger.debug(islp.getId() + ":" + islp.getParameter() + ":" + islp.getSchemeId() + ":" + islp.getSecurityLevelId() + ":" + islp.getType());
+            IssueSecuritySchemeMemberInfo memberInfo = new IssueSecuritySchemeMemberInfo();
+            memberInfo.id = String.valueOf(islp.getId());
+            memberInfo.issueSecurityLevelId = String.valueOf(islp.getSecurityLevelId());
+            JiraPermissionHolderInfo holder = new JiraPermissionHolderInfo();
+            holder.type = islp.getType();
+            memberInfo.holder = holder;
+            switch (islp.getType()) {
+              case "applicationRole":
+                break;
+              case "user":
+                // TODO: Set accountId or name? Set active based on api call?
+                holder.user = new AtlassianUser();
+                holder.user.key = islp.getParameter();
+                holder.user.active = true;
```

Comment:
I doubt we care if this is set for our purposes. Probably the right thing to do is look up the user in the user manager. Fine to leave as a TODO, since we won't use it.

Reply:
moved this code out of the plugin.